core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -428,21 +494,71 @@ private[sql] final case class ColumnTableScan(
           // initialize the decoder and store the starting cursor position
           $cursor = $decoder.initialize($buffer, $planSchema.apply($index));
         """)
-      bufferInitCode.append(
-        s"""
+
+
+      if (isWideSchema) {
+        if (bufferInitCode.length > 1024) {
+          bufferInitCodeBlocks.append(bufferInitCode.toString())
+          bufferInitCode.clear()
+        }
+
+        bufferInitCode.append(
+          s"""
+          final $decoderClass $decoderVar = $decoder;
+          $bufferVar = ($buffer == null || $buffer.isDirect()) ? null
```

Comment:
I will be chunking the whole string into a method in case of wide-schema. note the code where I clear bufferInitCode ( bufferInitCode.clear). Moving code outside the block will break the local variables of the created code.